src/effects/VST3/VST3Utils.cpp

```diff
@@ -60,16 +63,133 @@ wxString VST3Utils::MakeAutomationParameterKey(const Steinberg::Vst::ParameterIn
    return wxString::Format("%lu", static_cast<unsigned long>(parameterInfo.id));
 }
 
-bool VST3Utils::ParseAutomationParameterKey(const wxString& key, Steinberg::Vst::ParamID& paramId)
+
+const std::vector<VST3Wrapper::ParameterInfo>* VST3Wrapper::getParameterInfos() const
 {
-   const auto pos = key.Find('_');
-   const auto idStr = pos == wxNOT_FOUND ? key : key.Left(pos);
-   unsigned long value { };
-   if(idStr.ToULong(&value))
+   if (mParameterInfos.size() == 0)
    {
-      paramId = static_cast<Steinberg::Vst::ParamID>(value);
-      return true;
+      if (mEditController == nullptr)
+         return nullptr;
+
+      for (int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
```

Comment:
I have learned to prefer `size_t` not `int` for loops like this.

Reply:
ok, done